src/h3lib/lib/geoCoord.c

```diff
@@ -136,47 +136,39 @@ double constrainLng(double lng) {
 }
 
 /**
- * Find the great circle distance in radians between two spherical coordinates.
+ * The great circle distance in radians between two spherical coordinates.
  *
- * @param p1 The first spherical coordinates.
- * @param p2 The second spherical coordinates.
- * @return The great circle distance in radians between p1 and p2.
+ * This function uses the Haversine formula.
+ * For math details, see:
+ *     https://en.wikipedia.org/wiki/Haversine_formula
+ *     https://www.movable-type.co.uk/scripts/latlong.html
+ *
+ * @param  a  the first lat/lng pair (in radians)
+ * @param  b  the second lat/lng pair (in radians)
+ *
+ * @return    the great circle distance in radians between a and b
  */
-double _geoDistRads(const GeoCoord* p1, const GeoCoord* p2) {
-    // use spherical triangle with p1 at A, p2 at B, and north pole at C
-    double bigC = fabs(p2->lon - p1->lon);
-    if (bigC > M_PI)  // assume we want the complement
-    {
-        // note that in this case they can't both be negative
-        double lon1 = p1->lon;
-        if (lon1 < 0.0L) lon1 += 2.0L * M_PI;
-        double lon2 = p2->lon;
-        if (lon2 < 0.0L) lon2 += 2.0L * M_PI;
+double H3_EXPORT(pointDistRads)(const GeoCoord *a, const GeoCoord *b) {
+    double sinLat = sin((b->lat - a->lat) / 2);
+    double sinLng = sin((b->lon - a->lon) / 2);
```

Comment:
nit: prefer `/ 2.0`? I would write the double constants in division that way to avoid accidentally using integer division (not an issue where because `lat` and lon` are doubles, but still)